### DISCUSSION TITLE
Can you only serve one model per port?

### DISCUSSION THREAD

**Discussion by khari998** [mlflow/mlflow#12588] [Q&A] [answered]
I am trying to setup my inference server and I can't tell if I can only setup one model per port or if I can setup multiple models at the same port and query the invocations by the model name. 

For those with setups that require a lot of different models it would be useful to set them all up on one port or split them at a given port based on some underlying metadata dictating how the models are related.

Also is it possible to simply serve the models in the model registry? I saw that the staging/prod toggles were deprecated but it would've been nice to just set a model to staging and it was automatically deployed as serving on the default 5000 port and we could just query which models we want on the `invocations/` route by the model name. It seems like this would be much easier than using a separate port per each deployed model

**Comment by B-Step62** [comment, marked as answer]
(For others' reference) The short answer is MLflow doesn't support serving multiple models via single endpoint. For more context, please refer to the discussion here: https://github.com/mlflow/mlflow/issues/12589
